src/components/dbCalls.js

```diff
@@ -284,17 +284,17 @@ const killPlayerForRoom = async (target, roomID) => {
 //add player to database
 const addPlayerForRoom = async (player, roomID) => {
     const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
-    const trimmedLowercaseName  = player.trim().toLowerCase();
+    const trimmedLowercaseName  = player.replace(/\s/g, '').toLowerCase();
     //check if player already exists
-    const playerQuery = query(playerCollectionRef, where('name', '==', trimmedLowercaseName));
+    const playerQuery = query(playerCollectionRef, where('trimmedNameLowerCase', '==', trimmedLowercaseName));
     const playerSnapshot = await getDocs(playerQuery);
     if (!playerSnapshot.empty) {
         throw new Error('Player already exists');
     }
     //adds if not
     addDoc(playerCollectionRef, {
         name: player,
-        nameLowerCase: player.toLowerCase(),
+        trimmedNameLowerCase: player.replace(/\s/g, '').toLowerCase(),
```

Comment:
Ah regex. Does this just remove all spaces or how does it work?

Reply:
Yeah, it's to check if there are extra spaces in-between